nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -158,17 +191,43 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             final MongoCursor<Document> cursor = it.iterator();
             try {
                 FlowFile flowFile = null;
-                while (cursor.hasNext()) {
-                    flowFile = session.create();
-                    flowFile = session.write(flowFile, new OutputStreamCallback() {
-                        @Override
-                        public void process(OutputStream out) throws IOException {
-                            IOUtils.write(cursor.next().toJson(), out);
+                if (context.getProperty(RESULTS_PER_FLOWFILE).isSet()) {
+                    int ceiling = context.getProperty(RESULTS_PER_FLOWFILE).asInteger();
+                    List<Document> batch = new ArrayList<>();
+
+                    while (cursor.hasNext()) {
+                        batch.add(cursor.next());
+                        if (batch.size() == ceiling) {
+                            try {
+                                getLogger().info("Writing batch...");
```

Comment:
I would not log this at info level but would use debug instead (otherwise, it could be really verbose when the processor is used at a high rate).